pkg/apis/ocs/v1/storagecluster_types.go

```diff
@@ -106,6 +106,12 @@ type StorageDeviceSet struct {
 	// +optional
 	Replica int `json:"replica,omitempty"`
 
+	// DeviceType is the value of device type in
+	// this StorageDeviceSet. It can have only
+	// three values (SSD, HDD, NVMe)
+	// +optional
+	DeviceType string `json:"deviceType,omitempty"`
```

Comment:
```suggestion
	// DeviceType is the value of device type in
	// this StorageDeviceSet. It can have only
	// three values (SSD, HDD, NVMe)
	// +kubebuilder:validation:Enum=SSD;ssd;HDD;hdd;NVMe;NVME;nvme
	// +optional
	DeviceType string `json:"deviceType,omitempty"`
```

This will validate user input at client side.